velox/exec/Task.cpp

```diff
@@ -2736,6 +2747,25 @@ ContinueFuture Task::requestPause() {
   return makeFinishFutureLocked("Task::requestPause");
 }
 
+bool Task::pauseRequested(ContinueFuture* future) {
+  if (FOLLY_LIKELY(future == nullptr)) {
+    // Once 'pauseRequested_' is set, it will not be cleared until
+    // task::resume(). It is therefore OK to read it without a mutex
+    // from a thread that this flag concerns.
+    return pauseRequested_;
+  }
+
+  std::lock_guard<std::timed_mutex> l(mutex_);
+  if (!pauseRequested_) {
+    VELOX_CHECK(resumePromises_.empty())
```

Comment:
```
VELOX_CHECK(resumePromises_.empty());
```

Reply:
BTW, is this by some kind of coding convention of Velox? I could help add a compile time check for it to save some time for code reviewing.